internal/server/websocketServer.go

```diff
@@ -215,11 +219,63 @@ func (s *WebsocketServer) WebsocketHandler(w http.ResponseWriter, r *http.Reques
 			continue
 		}
 
-		// Write response
-		writeResponse(conn, respBytes)
+		// Write response with retries for transient errors
+		if err := writeResponseWithRetries(conn, respBytes, 3); err != nil {
+			s.logger.Error(fmt.Sprintf("Error reading message: %v", err))
+			break // Exit the loop on write error
+		}
 	}
 }
 
+func writeResponseWithRetries(conn *websocket.Conn, text []byte, maxRetries int) error {
```

Comment:
@ayushsatyam146 please add test cases around this. May be we can mock the websocket conn for different scenarios using `testify/mock`.